### PR TITLE
[CHANGELOG] Prepare v4.0.0-alpha.12

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,35 @@
...
+
+- [DataGrid] Add Density selector (#606) @DanailH
+- [DataGrid] Do not swallow keyboard events (#673) @DanailH
+- [DataGrid] Fix collision with react-virtualized on detectElementResize (#678) @tifosiblack
+- [DataGrid] Fix component name, rm context,  refact gridComponent (#707) @dtassone
+- [DataGrid] Fix infinite loop with multiple grid, and fix performance (#679) @dtassone
+- [DataGrid] Fix keyboard navigation in column picker (#674) @oliviertassinari
+- [DataGrid] Fix server-side sorting (#704) @akandels
+- [DataGrid] Improve the DX of popups (#686) @oliviertassinari
+- [DataGrid] Refactor cols (#682) @dtassone
```
Comment (DanailH) [inline CHANGELOG.md:20]: Should that be in the Core section, or does it change the api as well ?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,35 @@
...
+- [DataGrid] Rename hideToolbar prop to showToolbar (#706) @DanailH
+- [DataGrid] Server filters (#649) @dtassone
+- [DataGrid] Simplify footer logic (#676) @oliviertassinari
```
Comment (DanailH) [inline CHANGELOG.md:23]: Core section?

**Review by DanailH — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,35 @@
...
+- [DataGrid] Do not swallow keyboard events (#673) @DanailH
+- [DataGrid] Fix collision with react-virtualized on detectElementResize (#678) @tifosiblack
+- [DataGrid] Fix component name, rm context,  refact gridComponent (#707) @dtassone
+- [DataGrid] Fix infinite loop with multiple grid, and fix performance (#679) @dtassone
+- [DataGrid] Fix keyboard navigation in column picker (#674) @oliviertassinari
+- [DataGrid] Fix server-side sorting (#704) @akandels
+- [DataGrid] Improve the DX of popups (#686) @oliviertassinari
+- [DataGrid] Refactor cols (#682) @dtassone
+- [DataGrid] Rename hideToolbar prop to showToolbar (#706) @DanailH
+- [DataGrid] Server filters (#649) @dtassone
```
Comment (oliviertassinari) [inline CHANGELOG.md:22]: As it's not documented

```suggestion
- [DataGrid] Prepare server filters (#649) @dtassone
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,35 @@
...
+
+### Docs
+
+- [docs] Enable codesandbox preview in PRs (#613) @oliviertassinari
+
+### Core
+
+- [core] Batch small changes (#683) @oliviertassinari
+- [test] Add regression test (#705) @oliviertassinari
+- [test] Allow running all the tests in strict mode (#684) @oliviertassinari"
```
Comment (oliviertassinari) [inline CHANGELOG.md:33]:
```suggestion
- [test] Allow running all the tests in strict mode (#684) @oliviertassinari
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,35 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## [4.0.0-alpha.12](https://github.com/mui-org/material-ui-x/compare/v4.0.0-alpha.11...v4.0.0-alpha.12) (Dec 9, 2020)
+
```
Comment (oliviertassinari) [inline CHANGELOG.md:7]:
```suggestion

Big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:

- 🔍 Add a new data grid [density selector](https://material-ui.com/components/data-grid/rendering/#density) feature (#606) @DanailH.
- 💄 A first iteration on the data grid's toolbar.
- 🧪 Continue the iteration on the filtering feature, soon to be released @dtassone.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,35 @@
...
+### @material-ui/x-grid@v4.0.0-alpha.12 / @material-ui/data-grid@v4.0.0-alpha.12
+
+#### Changes
+
+- [DataGrid] Add Density selector (#606) @DanailH
+- [DataGrid] Do not swallow keyboard events (#673) @DanailH
```
Comment (oliviertassinari) [inline CHANGELOG.md:13]:
```suggestion
- [DataGrid] Fix swallowing of keyboard events (#673) @DanailH
```

File: CHANGELOG.md (hunk 2 above)
Comment (oliviertassinari) [reply CHANGELOG.md:23]: I would say it's a fix:
```suggestion
- [DataGrid] Fix display of selected rows in footer (#676) @oliviertassinari
```

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**